Pcap++/src/PcapRemoteDeviceList.cpp

```diff
@@ -48,19 +48,40 @@ namespace pcpp
 		}
 	}
 
+PcapRemoteDeviceList::PcapRemoteDeviceList(const IPAddress& ipAddress, uint16_t port, std::shared_ptr<PcapRemoteAuthentication> remoteAuth, std::vector<PcapRemoteDevice*> deviceList)
+	: m_RemoteDeviceList(std::move(deviceList))
+	, m_RemoteMachineIpAddress(ipAddress)
+	, m_RemoteMachinePort(port)
+	, m_RemoteAuthentication(std::move(remoteAuth))
+{}
+
 PcapRemoteDeviceList* PcapRemoteDeviceList::getRemoteDeviceList(const IPAddress& ipAddress, uint16_t port)
 {
-	return PcapRemoteDeviceList::getRemoteDeviceList(ipAddress, port, NULL);
+	auto result = PcapRemoteDeviceList::createRemoteDeviceList(ipAddress, port);
+	return result.release();
+}
+
+std::unique_ptr<PcapRemoteDeviceList> PcapRemoteDeviceList::createRemoteDeviceList(const IPAddress& ipAddress, uint16_t port)
+{
+	return PcapRemoteDeviceList::createRemoteDeviceList(ipAddress, port, nullptr);
 }
 
 PcapRemoteDeviceList* PcapRemoteDeviceList::getRemoteDeviceList(const IPAddress& ipAddress, uint16_t port, PcapRemoteAuthentication* remoteAuth)
 {
+	auto result = PcapRemoteDeviceList::createRemoteDeviceList(ipAddress, port, remoteAuth);
+	return result.release();
+}
+
+std::unique_ptr<PcapRemoteDeviceList> PcapRemoteDeviceList::createRemoteDeviceList(const IPAddress& ipAddress, uint16_t port, PcapRemoteAuthentication const* remoteAuth)
+{
+	std::shared_ptr<PcapRemoteAuthentication> sRemoteAuth;
```

Comment:
I don't like the naming `sRemoteAuth`, it's not the convention in this project.

Reply:
however, we don't have many `std::shared_ptr` currently.  So if other people agree with this, this can become the official coding style.

Reply:
Originally, I had a private overload where `remoteAuth` was a shared ptr to avoid that, but scrapped it as it was not strictly nessesary. 

Perhaps that would be a better solution?

Reply:
Tbh, I just named it that way coz i needed the variable to not conflict. If there are suggestions on a potential name, i'm happy to hear them.

Reply:
Changed it to `remoteAuthCopy` to be more explicit that its a copy in 4f66704. Not sure if it should be `remoteAuthCopy` or `pRemoteAuthCopy` tho.

Reply:
I am fine with either, but seems `pXXX` is more common in the codebase?

Reply:
Updated cf3d1c2